csharp-basic-topics/GenerateRandomColorName/GenerateRandomColorName/KnownColorGenerator.cs

```diff
@@ -0,0 +1,20 @@
+namespace GenerateRandomColorName;
+using System.Drawing;
+
+public class KnownColorGenerator
+{
+    private Random randomGen;
+
+    public KnownColorGenerator()
+    {
+        randomGen = new Random();
+    }
+
+    public Color GetRandomKnownColor()
+    {
+        KnownColor[] allColors = (KnownColor[])Enum.GetValues(typeof(KnownColor));
```

Comment:
Here we can use the generic overload [Enum.GetValues<TEnum>()](https://learn.microsoft.com/en-us/dotnet/api/system.enum.getvalues?view=net-8.0#system-enum-getvalues-1):
`Enum.GetValues<KnownColor>();`
Also, since this is something that is constant, we can make it a static readonly member of the class and just create the array once.
We also don't need to have `Random` as a member since we are not seeding it. We can just use `Random.Shared`. This means that the whole class now can be static and just have a single static `GetRandomKnownColor()` method.